website/content/api-docs/secret/databases/snowflake.mdx

```diff
@@ -90,9 +90,13 @@ list the plugin does not support that statement type.
   statements executed to create and configure a user. Must be a
   semicolon-separated string, a base64-encoded semicolon-separated string, a
   serialized JSON string array, or a base64-encoded serialized JSON string
-  array. The `{{name}}`, `{{password}}` and `{{expiration}}` values will be
-  substituted. The generated password will be a random alphanumeric 20 character
-  string.
+  array. The `{{name}}` and `{{expiration}}` values will be substituted.
+
+  The following values will be substituted depending on the
+  [credential_type](/api-docs/secret/databases#credential_type) of the role:
+
+  - `{{password}}` is substituted for the `password` credential type
+  - `{{public_key}}` is substituted for the `rsa_private_key` credential type
```

Comment:
I guess this could be a partial too. Not sure if it's worth it though. Pretty small section that isn't likely to be added to given that `{{public_key}}` is pretty specific to Snowflake.